soc/arm/silabs_exx32/common/soc.c

```diff
@@ -230,4 +250,13 @@ static int silabs_exx32_init(const struct device *arg)
 	return 0;
 }
 
+#if defined(CONFIG_SOC_GECKO_DEV_INIT) && defined(CONFIG_PM)
+void init_hfxo_irq(void)
+{
+	IRQ_DIRECT_CONNECT(HFXO0_IRQn, 2, sl_hfxo_manager_irq_handler, 0);
```

Comment:
Can we add the interrupt info to the devicetree here and then use the `DT_INST_IRQN` and `DT_INST_IRQN` macros?

Reply:
I've refactored the code and this piece of code is no longer needed.